### PR TITLE
Php 7

### PR DESCRIPTION
**Pull request by emanuele45**
A set of changes to make Elk a bit more php-7-ready.

I have just few doubts:
  1. substr() now returns an empty string, if string is equal to start characters long. 
  2. the eval in Cache.class.php (I suspect it should be extracted from the if and put in a try/catch)
  3. the list comment <blockquote>In general, it is recommended not to rely on the order in which list() assignments occur, as this is an implementation detail that may change again in the future. </blockquote>
  4. #2599 
  5. the changes to foreach may create some subtle bug http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.foreach but review all the foreach is basically impossible, so let's hope to find them during beta (yeah, I like betting! :stuck_out_tongue_winking_eye:  )

and that's all.

### CODE REVIEW

**Comment by emanuele45** [comment]
I'm not sure if it is worth backporting these to 1.0.
If 1.1 development goes smooth enough, we may release 1.1 well before the end of the year. In that case I wouldn't bother.

File: sources/Debug.class.php [hunk 1]
```diff
@@ -181,14 +181,15 @@ public function get_db()
...
+	 * of the script execution and one at the end)
 	 *
 	 * @param string $point can be end or start depending on when the function
 	 *               is called
 	 * @param string[]|null $rusage value of getrusage or null to let the method call it
 	 */
 	public function rusage($point, $rusage = null)
 	{
+		// getrusage is missing in php < 7 on Windows
 		if (!function_exists('getrusage') || !$this->_track)
```
Comment (Spuds) [inline sources/Debug.class.php]: NitPik I'd reverse this order .. no need to do the function exists if you are not tracking to begin with.

**Comment by Spuds** [comment]
I suppose its fine to use this in 1.0.8 ( I don';t see any harm ), but until we know more we can't claim php7 compatibility.  Some of the changes like list(,,$bla) no longer working and the foreach by value change description reads really oddly.

**Comment by emanuele45** [comment]
Yep, yep indeed.

File: sources/Debug.class.php (hunk 1 above)
Comment (emanuele45) [inline sources/Debug.class.php]: done!

**Comment by live627** [comment]
> foreach by value change description reads really oddly.

Eh?

**Comment by emanuele45** [comment]
Well, I guess it can be merged. :angel:

**Pull request merged**